system-build.sh

```diff
@@ -248,12 +248,7 @@ for T in $LLVM_TOOLS; do
 		exitmsg "Cannot find working $T binary".
 	fi
 
-	if readlink -- "${TOOL}" >/dev/null; then
-		# avoid making copy of a ccache executable (symlink's target)
-		ln -fs "${TOOL}" "${LLVM_PREFIX}/bin/"
-	else
-		cp ${TOOL} $LLVM_PREFIX/bin
```

Comment:
Doesn't this break the `full-archive` option? That is, will git copy the targets of symbolic links into the archive? (Probably yes, just wondering).

Reply:
I've just tried the `full-archive` option and `git archive` preserves symlinks. `system-build.sh` uses system libraries, so this change should not affect portability (which is bounded by the same os (and distro) and its release anyway).

```
$ unzip symbiotic.zip
...
 extracting: symbiotic/properties/valid-memcleanup.prp  
  inflating: symbiotic/properties/valid-memsafety.prp  
finishing deferred symbolic links:
  symbiotic/llvm-9.0.1/bin/clang -> /usr/lib64/ccache/clang
  symbiotic/llvm-9.0.1/bin/llvm-dis -> /usr/bin/llvm-dis
  symbiotic/llvm-9.0.1/bin/llvm-link -> /usr/bin/llvm-link
  symbiotic/llvm-9.0.1/bin/llvm-nm -> /usr/bin/llvm-nm
  symbiotic/llvm-9.0.1/bin/opt -> /usr/bin/opt
...
```

Also, I have just spotted, that the following system dependencies are unnecessarily copied to the archive (on Fedora 31) in `push-to-git.sh`:
* `lib/libstdc++.so.6`
* `lib/libtinfo.so.6`
* `lib/libz3.so.0`

Reply:
Those dependencies are there on purpose and only with the `full-archive` option. If you use just `archive`, those should not be included (if not in the git already).

Reply:
I have updated the commit to respect the `full-archive` option. However, the archive will never we portable as `clang` is just a symlink to `clang-$version` and the `libLLVM-$version.so` (and probably other shared libraries too) is not included in the final zip archive.